src/api/catalogueItem.tsx

```diff
@@ -191,3 +194,56 @@ export const useEditCatalogueItem = (): UseMutationResult<
     }
   );
 };
+
+export const useMoveToCatalogueItem = (): UseMutationResult<
+  TransferState[],
+  AxiosError,
+  MoveToCatalogueItem
+> => {
+  const queryClient = useQueryClient();
+  return useMutation(async (moveToCatalogueItem: MoveToCatalogueItem) => {
+    const transferStates: TransferState[] = [];
+    let successfulIds: string[] = [];
+
+    const promises = moveToCatalogueItem.selectedItems.map(
+      async (item: EditCatalogueItem, index) => {
+        return editCatalogueItem({
+          id: item.id,
+          catalogue_category_id:
+            moveToCatalogueItem.targetCatalogueCategory?.id,
+        })
+          .then((result) => {
+            transferStates.push({
+              name: result.name ?? '',
+              message: `Successfully moved to ${
+                moveToCatalogueItem.targetCatalogueCategory?.name || 'Root'
+              }`,
+              state: 'success',
+            });
+            successfulIds.push(item.id);
+          })
+          .catch((error) => {
+            const response = error.response?.data as ErrorParsing;
+
+            transferStates.push({
+              name: moveToCatalogueItem.selectedItems[index].name,
+              message: response.detail,
+              state: 'error',
+            });
+          });
+      }
+    );
+
+    await Promise.all(promises);
+
+    if (successfulIds.length > 0) {
+      queryClient.invalidateQueries({ queryKey: ['CatalogueItems'] });
+      queryClient.invalidateQueries({ queryKey: ['CatalogueBreadcrumbs'] });
+      successfulIds.map((id: string) =>
```

Comment:
```suggestion
      successfulIds.forEach((id: string) =>
```
Very minor - but we don't need it to return a new array, so forEach should be slightly faster.